src/java/test/org/apache/zookeeper/test/ReadOnlyModeTest.java

```diff
@@ -251,14 +257,19 @@ public void testSessionEstablishment() throws Exception {
     @Test(timeout = 90000)
     public void testSeekForRwServer() throws Exception {
         // setup the logger to capture all logs
-        Layout layout = Logger.getRootLogger().getAppender("CONSOLE")
-                .getLayout();
+        LoggerContext loggerContext =  (LoggerContext) LogManager.getContext(false);
+        Configuration logConfig = loggerContext.getConfiguration();
+        
+        Layout layout =
+        		logConfig.getRootLogger().getAppenders().get("STDOUT").getLayout();
+        String appenderName = "zkAppender";
         ByteArrayOutputStream os = new ByteArrayOutputStream();
-        WriterAppender appender = new WriterAppender(layout, os);
-        appender.setImmediateFlush(true);
-        appender.setThreshold(Level.INFO);
-        Logger zlogger = Logger.getLogger("org.apache.zookeeper");
-        zlogger.addAppender(appender);
+        WriterAppender appender = WriterAppender.createAppender((StringLayout)layout, null, new OutputStreamWriter(os), appenderName, false, true);
```

Comment:
It's probably a good idea to use the builder to create the appender (see above).